tensorflow/contrib/client/python/framework/session_run.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2016 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from tensorflow.python.client import session
+
+def register_session_fetch_feed_conversion_functions(tensor_type,
+    fetch_function, feed_function=None, feed_function_for_partial_run=None):
+  """ Register fetch and feed conversion functions for session.run.
+  This function is for registering fetch and feed conversion functions for
+  user defined type which may be passed to session.run as fetch or feed params.
```

Comment:
This function registers a triple of conversion functions for fetching and/or feeding values of user-defined types in a call to `tf.Session.run()`.